e3db/types/note_options.py

```diff
@@ -0,0 +1,81 @@
+from datetime import date
+
+# Look for a possible alternative to a NoteOptions class. Not a huge priority at the moment, 
+# but this may not be what is wanted. 
+class NoteOptions():
+    
+    def __init__(self, client_id: str, max_views: int, note_name: str, expiration: date, # Check `date` type for expiration. 
+                expires: bool, type: str, plain: dict, file_meta: dict, eacp=None):
+        """
+        Initializes the NoteOptions class. Represents optional values not necessary
+        for creating a note. 
+
+        Parameters
+        ----------
+        client_id : str
+
+        max_views : int
+
+        note_name : str
+
+        expiration : Date
+
+        expires : bool
+
+        eacp : EACP 
+
+        type : str
+
+        plain : dict[string]string
+
+        file_meta : dict[string]string 
+
+        Returns
+        -------
+        None
+        """
+
+        # These may need to be typed properly! 
+
+        # Premium options
+        self.__client_id     = client_id
+        self.__max_views     = max_views
+        self.__expiration    = expiration
+        self.__expires       = expires
+        self.__eacp          = eacp
+
+        # Non-premium option
+        self.__type          = type
+        self.__plain         = plain
+        self.__file_meta     = file_meta
+
+    def to_json(self) -> dict:
+        """
+        Seriaizes the note as a JSON-style object.
+
+        Parameters
+        ----------
+        None
+
+        Returns
+        -------
+        dict
+            JSON-style document containing the Note elements
+        """
+        to_serialize = {
+            'client_id': self.__client_id,
+            'max_views': self.__max_views,
+            'note_name': self.__note_name,
+            'expiration': self.__expiration,
+            'expires': self.__expires,
+            'eacp': self.__eacp,
+            'type': self.__type,
+            'plain': self.__plain,
+            'file_meta': self.__file_meta
+        }
+
+        for key, value in list(to_serialize.items()):
```

Comment:
While I think there is an earlier comment to the effect that this isn’t needed, if it was consider the value of extracting this out to a method in a utils class that could be used across all the places this code block appears, and if we ever needed to update the logic for which `None`  Keys should be dropped (or which should cause a helpful exception to be thrown or logged - not as likely for what this code does but in general this is an example of changes to the a codebase that can be done with less future effort given investment in a abstractions like functions for re-usable logic.)